### PR TITLE
python312Packages.pytorch-bench: init at unstable-2024-07-18

### PR DESCRIPTION
**Pull request by drupol**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by drupol** [comment]
@ofborg build python312Packages.torch-benchmark

**Comment by drupol** [comment]
@ofborg build python312Packages.torch-benchmark

**Comment by drupol** [comment]
@ofborg build python312Packages.torch-benchmark

**Comment by drupol** [comment]
@ofborg build python312Packages.pytorch-bench

File: pkgs/development/python-modules/torchprofile/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+buildPythonPackage rec {
+  pname = "torchprofile";
+  version = "0.0.3";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "zhijian-liu";
+    repo = "torchprofile";
+    rev = "v${version}";
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/torchprofile/default.nix:19]:
```suggestion
    rev = "refs/tags/v${version}";
```

File: pkgs/development/python-modules/torchprofile/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  ];
+
+  pythonImportsCheck = [
+    "torchprofile"
+  ];
+
+  meta = {
+    description = "A general and accurate MACs / FLOPs profiler for PyTorch models";
+    homepage = "https://github.com/zhijian-liu/torchprofile";
+    license = lib.licenses.mit;
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/torchprofile/default.nix:44]:
```suggestion
    homepage = "https://github.com/zhijian-liu/torchprofile";
    changelog = "https://github.com/zhijian-liu/torchprofile/releases/tag/v${version}";
    license = lib.licenses.mit;
```

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
minor details.

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 352827`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python311Packages.pytorch-bench</li>
    <li>python311Packages.pytorch-bench.dist</li>
    <li>python311Packages.torchprofile</li>
    <li>python311Packages.torchprofile.dist</li>
    <li>python312Packages.pytorch-bench</li>
    <li>python312Packages.pytorch-bench.dist</li>
    <li>python312Packages.torchprofile</li>
    <li>python312Packages.torchprofile.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python311Packages.pytorch-bench</li>
    <li>python311Packages.pytorch-bench.dist</li>
    <li>python311Packages.torchprofile</li>
    <li>python311Packages.torchprofile.dist</li>
    <li>python312Packages.pytorch-bench</li>
    <li>python312Packages.pytorch-bench.dist</li>
    <li>python312Packages.torchprofile</li>
    <li>python312Packages.torchprofile.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python311Packages.pytorch-bench</li>
    <li>python311Packages.pytorch-bench.dist</li>
    <li>python311Packages.torchprofile</li>
    <li>python311Packages.torchprofile.dist</li>
    <li>python312Packages.pytorch-bench</li>
    <li>python312Packages.pytorch-bench.dist</li>
    <li>python312Packages.torchprofile</li>
    <li>python312Packages.torchprofile.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python311Packages.pytorch-bench</li>
    <li>python311Packages.pytorch-bench.dist</li>
    <li>python311Packages.torchprofile</li>
    <li>python311Packages.torchprofile.dist</li>
    <li>python312Packages.pytorch-bench</li>
    <li>python312Packages.pytorch-bench.dist</li>
    <li>python312Packages.torchprofile</li>
    <li>python312Packages.torchprofile.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/torchprofile/default.nix (hunk 1 above)
Comment (drupol) [reply pkgs/development/python-modules/torchprofile/default.nix:19]: Ah thank you, I always forgot that one.

File: pkgs/development/python-modules/torchprofile/default.nix (hunk 2 above)
Comment (drupol) [reply pkgs/development/python-modules/torchprofile/default.nix:44]: Fixed!

File: pkgs/development/python-modules/torchprofile/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    torchvision
+  ];
+
+  pythonImportsCheck = [
+    "torchprofile"
+  ];
+
+  meta = {
+    changelog = "https://github.com/zhijian-liu/torchprofile/releases/tag/v${version}";
+    description = "A general and accurate MACs / FLOPs profiler for PyTorch models";
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/torchprofile/default.nix:43]:
```suggestion
    description = "General and accurate MACs / FLOPs profiler for PyTorch models";
```
https://nixos.org/manual/nixpkgs/unstable/#var-meta-description

Sorry :')

Comment (drupol) [reply pkgs/development/python-modules/torchprofile/default.nix]: Fixed, thanks !

**Review by GaetanLepage — APPROVED** [review]
All good !

**Comment by drupol** [comment]
@NixOS/nixpkgs-merge-bot merge

**Comment by GaetanLepage** [comment]
Let's maybe wait for ogborg

**Pull request merged**